validator/src/main/java/com/google/daq/mqtt/util/ConfigManager.java

```diff
@@ -32,25 +39,93 @@
 /**
  * Container class for working with generated UDMI configs (from device metadata).
  */
-public class ConfigGenerator {
+public class ConfigManager {
 
   public static final String GENERATED_CONFIG_JSON = "generated_config.json";
   public static final ProtocolFamily DEFAULT_FAMILY = ProtocolFamily.VENDOR;
-
+  private static final String CONFIG_DIR = "config";
   private final Metadata metadata;
+  private final String deviceId;
+  private final SiteModel siteModel;
 
-  public ConfigGenerator(Metadata metadata) {
+  /**
+   * Initiates ConfigManager for the given device at the given file location.
+   *
+   * @param metadata Device metadata
+   * @param deviceId Device ID
+   * @param siteDir Pathe to site model
+   */
+  public ConfigManager(Metadata metadata, String deviceId, SiteModel siteModel) {
     this.metadata = metadata;
+    this.deviceId = deviceId;
+    this.siteModel = siteModel;
+  }
+
+  public static ConfigManager configFrom(Metadata metadata) {
+    return new ConfigManager(metadata, null, null);
+  }
+
+  public static ConfigManager configFrom(Metadata metadata, String deviceId, SiteModel siteModel) {
+    return new ConfigManager(metadata, deviceId, siteModel);
+  }
+
+  private boolean isStaticConfig() {
+    return catchToNull(() -> metadata.cloud.config.static_file) != null;
+  }
+
+  private String readStaticConfigFromFile(String fileName) {
+    try {
+      File deviceDir = siteModel.getDeviceDir(deviceId);
+      File configDir = new File(deviceDir, CONFIG_DIR);
+      File configFile = new File(configDir, fileName);
+      String canonicalPath = configFile.getCanonicalPath();
+      String configDirPath = configDir.getCanonicalPath();
+      if (!canonicalPath.startsWith(configDirPath)) {
+        throw new IllegalArgumentException();
+      }
+      return readFileToString(configFile, StandardCharsets.UTF_8);
+    } catch (Exception e) {
+      throw new RuntimeException(String.format("While reading config file: %s", fileName), e);
+    }
+  }
+
+  public Config deviceConfig() {
+    return generateDeviceConfig();
   }
 
-  public static ConfigGenerator configFrom(Metadata metadata) {
-    return new ConfigGenerator(metadata);
+  /**
+   * Whether the device config be downgraded.
+   *
+   * @return yes/no
+   */
+  public boolean canBeDowngraded() {
+    return ! isStaticConfig();
+  }
+
+  /**
+   * Returns a JSON representation of the device config, allowing for any static config overrides.
+   *
+   * @return JSON Object of Device Config
+   */
+  public JsonNode deviceConfigJson() {
+    if (isStaticConfig()) {
+      String staticFileContents = readStaticConfigFromFile(metadata.cloud.config.static_file);
+      try {
+        return OBJECT_MAPPER_STRICT.readTree(staticFileContents);
```

Comment:
There code in JsonUtils that does most of this.  You should just be able to compute and calcualte the "configFile" and then pass that into the utility function.  The "readFileToString" and "OBJECT_MAPPER_STRICT parts should be encapsulated in JsonUtil already.